pkg/template/controller/readiness.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/apis/batch"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	deploymentutil "k8s.io/kubernetes/pkg/controller/deployment/util"
+
+	deployapi "github.com/openshift/origin/pkg/deploy/apis/apps"
+)
+
+func checkDeploymentReadiness(obj runtime.Object) (bool, bool) {
```

Comment:
@smarterclayton some follow up questions:

> I had expected a per api group helper upstream (dealing with both internal and unversioned) 

do you mean just a utility function, or an actual api endpoint?  Sounds like the former.

>  which exposed a condition that took a deployment / statefulset / daemonset and returned "ready" or "not ready". 

exposed a condition how?  is this a reference to api conditions, or just a general use of the term?  my current interpretation of this is "utility function that takes as inputs a desired condition and a resource object, and returns whether that object meets the desired condition or not".... this utility function would be implemented by various api groups for their resource types accurate?  Not sure who is responsible for looking up the utility function to call for a given Object.

>  And maybe a variant that took a client for services and fetched an endpoint or something.

so in this case the readiness check would be available as an api endpoint... associated w/ the api group presumably?

>  I then expected to layer a kubectl wait command around those basic libraries. Eventually, I expected to have a subresource for resources that would encapsulate whatever we learned from kubectl wait.

How much of this vision are you requesting be driven via this PR?  Or can this PR proceed as is with the intent of adapting it to the aforementioned helpers when they become available?